lib/github-pages-health-check/repository.rb

```diff
@@ -0,0 +1,70 @@
+module GitHubPages
+  module HealthCheck
+    class Repository < GitHubPages::HealthCheck::Checkable
+
+      attr_reader :name, :owner
+
+      REPO_REGEX = %r{\A[a-z0-9_\-]+/[a-z0-9_\-\.]+\z}i
+
+      def initialize(name_with_owner, access_token: nil)
+        unless name_with_owner =~ REPO_REGEX
+          raise GitHubPages::HealthCheck::Errors::InvalidRepositoryError
```

Comment:
For consistency with the rest of the class, this should either be `Errors::InvalidRepositoryError`, or all the others should have `GitHubPages::HealthCheck::` :smile: